jlm/rvsdg/operation.cpp

```diff
@@ -20,14 +20,14 @@ port::port(const jlm::rvsdg::type & type)
     : port(type.copy())
 {}
 
-port::port(std::unique_ptr<jlm::rvsdg::type> type)
+port::port(std::shared_ptr<const jlm::rvsdg::type> type)
     : type_(std::move(type))
 {}
 
 bool
 port::operator==(const port & other) const noexcept
 {
-  return *type_ == *other.type_;
+  return type_ == other.type_ || *type_ == *other.type_;
```

Comment:
This is a performance optimization, right? First checking for pointer in case of static instances, and then if that is false, invoking `operator==()` on the pointer. Could we put a comment on this expressing it?